src/Mvc/Mvc.Core/src/MvcOptions.cs

```diff
@@ -102,6 +103,25 @@ public MvcOptions()
         /// </summary>
         public FormatterCollection<IInputFormatter> InputFormatters { get; }
 
+        /// <summary>
+        /// Gets or sets a value that detemines if the inference of <see cref="RequiredAttribute"/> for
+        /// for properties and parameters of non-nullable reference types is suppressed. If <c>false</c> 
+        /// (the default), then all non-nullable reference types will behave as-if <c>[Required]</c> has 
+        /// been applied. If <c>true</c>, this behavior will be suppressed; nullable reference types and 
+        /// non-nullable reference types will behave the same for the purposes of validation.
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// This option controls whether MVC model binding and validation treats nullable and non-nullable 
+        /// reference types differently. 
+        /// </para>
+        /// <para>
+        /// By default, MVC will treat a non-nullable reference type parameters and properties as-if 
+        /// <c>[Required]</c> has been applied, resulting in validation errors when no value was bound.
+        /// </para>
+        /// </remarks>
+        public bool SuppressImplicitRequiredAttributeForNonNullableReferenceTypes { get; set; }
```

Comment:
I challenge anyone to come up with a better property name. This is a masterpiece.

Reply:
> I challenge anyone to come up with a better property name. This is a masterpiece.

Invert the true/false and call it `RequireValueForNonNullableReferences  { get; set; } = true`?

Reply:
nah, I want it to be false by default. I'm actually happy with the name.